src/sentry/sentry_metrics/kafka.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, Mapping, Optional, Sequence, Union
+
+import sentry_kafka_schemas
+from arroyo import Topic
+from arroyo.backends.abstract import Producer
+from arroyo.backends.kafka import KafkaPayload, KafkaProducer, build_kafka_configuration
+from django.conf import settings
+from django.core.cache import cache
+
+from sentry import quotas
+from sentry.sentry_metrics.base import GenericMetricsBackend
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+from sentry.utils import json
+from sentry.utils.kafka_config import get_kafka_producer_cluster_options
+
+ingest_codec: sentry_kafka_schemas.codecs.Codec[Any] = sentry_kafka_schemas.get_codec(
+    "ingest-metrics"
+)
+
+
+def build_mri(metric_name: str, type: str, use_case_id: UseCaseID, unit: Optional[str]) -> str:
+    mri_unit = "none" if unit is None else unit
+    return f"{type}:{use_case_id.value}/{metric_name}@{mri_unit}"
+
+
+def get_retention_from_org_id(org_id: int) -> int:
+    cache_key = f"seen-orgs:{org_id}"
```

Comment:
nit: I might change this prefix to describe what the data is e.g. `org_retention:{org_id}`

Reply:
fixed